src/main/java/csdev/couponstash/model/history/CommandTextHistory.java

```diff
@@ -1,63 +1,52 @@
 package csdev.couponstash.model.history;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.LinkedList;
 
 /**
  * This class stores all the commandText that was executed in STASH. Does not matter
  * if the commands had errors, they are still stored and can be retrieved by pressing up or down on
  * the keyboard.
  */
 public class CommandTextHistory {
-    private List<String> commandTextHistory;
+    private static final String EMPTY_COMMAND = "";
+
+    private LinkedList<String> commandTextHistory;
     private int currIndex;
 
     public CommandTextHistory() {
-        commandTextHistory = new ArrayList<>();
-        commandTextHistory.add("");
-        commandTextHistory.add("");
-        currIndex = 1;
+        commandTextHistory = new LinkedList<>();
```

Comment:
What a genius way to implement this history stack!

Reply:
All learn frm the best like master chester